### ISSUE TITLE
debian packaging: file conflict between buildah and containers common #2371 

### ISSUE THREAD

**Issue by tomtom94** [containers/buildah#4012]
Hello, I wanna reopen an issue closed few years ago 😂 https://github.com/containers/buildah/issues/2371

I just encountered the issue in my VM Ubuntu 20.04 LTS

Since last Friday I got that
```
Unpacking containers-common (100:1-22) ...
dpkg: error processing archive /tmp/apt-dpkg-install-HDkSYr/22-containers-common_100%3a1-22_all.deb (--unpack):
 trying to overwrite '/etc/containers/policy.json', which is also in package golang-github-containers-common 0.44.4+ds1-1
```

**3 days ago my VM was installing perfectly but I had to delete and recreate it 😞** since then the exact same install batch doesn't work for my exam online.

This is due to (if ever you wanna reproduce)
```
# Install cri-o
sudo apt-get update && sudo apt-get upgrade -y
export OS=xUbuntu_20.04
export VERSION=1.23
echo "deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/$VERSION/$OS/ /" | sudo tee -a /etc/apt/sources.list.d/cri-0.list
curl -L http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/$VERSION/$OS/Release.key | sudo apt-key add -
echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/ /" | sudo tee -a /etc/apt/sources.list.d/libcontainers.list
sudo apt-get update
sudo apt-get install -y cri-o cri-o-runc podman buildah
```

The last command line above will throw you this
```
Errors were encountered while processing:
 /tmp/apt-dpkg-install-HDkSYr/22-containers-common_100%3a1-22_all.deb
needrestart is being skipped since dpkg has failed
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Need help 😂

**Comment by vrothberg** [comment]
@lsm5 PTAL

**Comment by lsm5** [comment]
@tomtom94 hi, I've removed the buildah package from the Kubic repository you mentioned and I'll be removing podman from there today. I recommend switching to 22.04 LTS so that you can get buildah from the default ubuntu repos, or building buildah from source.

The kubic repos will continue to remain active for cri-o and its dependencies.

**Comment by tomtom94** [comment]
It doesn't work yet @lsm5 

```
Unpacking containers-common (100:1-22) ...
dpkg: error processing archive /tmp/apt-dpkg-install-P0vPIn/22-containers-common_100%3a1-22_all.deb (--unpack):
 trying to overwrite '/etc/containers/policy.json', which is also in package golang-github-containers-common 0.44.4+ds1-1
```
```
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 cri-o : Depends: containers-common (>= 0.1.27) but it is not going to be installed
 kubeadm : Depends: kubernetes-cni (>= 0.8.7)
           Depends: cri-tools (>= 1.19.0) but it is not going to be installed
 kubelet : Depends: kubernetes-cni (>= 0.8.7)
           Depends: socat but it is not going to be installed
           Depends: ebtables but it is not going to be installed
           Depends: conntrack but it is not going to be installed
```
I am using a batch given by The Linux Foundation to sit for the Kubernetes Certification, it had worked well last week 😞

**Comment by lsm5** [comment]
@haircommander can you take a look please? ^

**Comment by haircommander** [comment]
running in an ubuntu 20.04 container (after installing curl and gnupg) worked for me. I did not install `podman` and `buildah` as they weren't available. Maybe the repos just needed to refresh?

**Comment by tomtom94** [comment]
It doesn't work.

And the error is ever unknown in their issues https://github.com/cri-o/cri-o/issues?q=is%3Aissue+is%3Aopen+dpkg%3A+error+processing+archive

Moreover my installing batch is always using the same crio-o version, this is why I turned to you

```
export VERSION=1.23
echo "deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/$VERSION/$OS/ /" | sudo tee -a /etc/apt/sources.list.d/cri-0.list
curl -L http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/$VERSION/$OS/Release.key | sudo apt-key add -
```

Doesn't seem consistent to post an issue on cri-o repository 🥕

**Comment by mythi** [comment]
> running in an ubuntu 20.04 container (after installing curl and gnupg) worked for me. I did not install `podman` and `buildah` as they weren't available. Maybe the repos just needed to refresh?

@haircommander you’ll need 20.10 or later to reproduce (that was the first release buildah/podman were part of the distro)

**Comment by ccamacho** [comment]
@lsm5 quick question, why did you remove the packages so quickly? Things like juju only work with 20.04 so far removing those packages basically broke anyone deploying juju with Ubuntu 20.04, also anyone trying podman in GitHub actions is broken.

**Comment by lsm5** [comment]
> @lsm5 quick question, why did you remove the packages so quickly? Things like juju only work with 20.04 so far removing those packages basically broke anyone deploying juju with Ubuntu 20.04, also anyone trying podman in GitHub actions is broken.

- Those packages weren't being maintained
- The installation docs did discourage from using Kubic repos in production
- People were pinging me about the flaky opensuse mirror causing podman installation issues
- Podman v3.4.2 has 4 CVEs which we were not going to publish updates for on the Kubic repos

All these combined made me think I'd rather the users have no package at all than a flaky vulnerable package. I did comment on a podman issue that I'd remove the podman package few days later and I also had no idea at all buildah and skopeo were being used on a large scale through the Kubic repos.

Anyway, now I have realized the popularity of the Kubic repos and the packages are back now. I have checked with the GitHub actions people, and their 18.04 and 20.04 have been fixed with Kubic while 22.04 has now switched to container tools from the default ubuntu repos itself. AFAICT, all's well now. Sorry about the mess.

Also, I will be keeping those Kubic packages perhaps forever, given Ubuntu's 5-year (maybe 10) lifecycle and won't be providing any further updates now that everything's fixed.

I have some work in progress on a much more maintainable way of generating Debian packages at https://build.opensuse.org/project/show/devel:kubic:libcontainers:unstable . This is mainly being done RE: ubuntu testing on upstream container tools and I can't make any guarantees on this new repo. But if anyone's feeling adventurous, they're welcome to try.
